op-node/rollup/derive/kromampt_upgrade_transactions_test.go

```diff
@@ -53,67 +57,175 @@ func TestMPTSourcesMatchSpec(t *testing.T) {
-	require.Equal(t, uint64(50_000), updateL1BlockProxy.Gas())
-	require.Equal(t, common.FromHex("0x3659cfe6000000000000000000000000cd7467a8926d13f8b41ea035ff3761326c822bad"), updateL1BlockProxy.Data())
-
-	updateBaseFeeVaultProxySender, updateBaseFeeVaultProxy := toDepositTxn(t, upgradeTxns[5])
-	require.Equal(t, updateBaseFeeVaultProxySender, common.Address{})
-	require.Equal(t, updateBaseFeeVaultProxySource.SourceHash(), updateBaseFeeVaultProxy.SourceHash())
-	require.NotNil(t, updateBaseFeeVaultProxy.To())
-	require.Equal(t, *updateBaseFeeVaultProxy.To(), predeploys.BaseFeeVaultAddr)
-	require.Equal(t, uint64(50_000), updateBaseFeeVaultProxy.Gas())
-	require.Equal(t, common.FromHex("3659cfe6000000000000000000000000db78bab44d9632e68348659dd47b4806ba276d89"), updateBaseFeeVaultProxy.Data())
-
-	updateL1FeeVaultProxySender, updateL1FeeVaultProxy := toDepositTxn(t, upgradeTxns[6])
-	require.Equal(t, updateL1FeeVaultProxySender, common.Address{})
-	require.Equal(t, updateL1FeeVaultProxySource.SourceHash(), updateL1FeeVaultProxy.SourceHash())
-	require.NotNil(t, updateL1FeeVaultProxy.To())
-	require.Equal(t, *updateL1FeeVaultProxy.To(), predeploys.L1FeeVaultAddr)
-	require.Equal(t, uint64(50_000), updateL1FeeVaultProxy.Gas())
-	require.Equal(t, common.FromHex("3659cfe6000000000000000000000000d86e1a7c380f398bda3d598ee65c891ce5c3c8f0"), updateL1FeeVaultProxy.Data())
-
-	updateSequencerFeeVaultProxySender, updateSequencerFeeVaultProxy := toDepositTxn(t, upgradeTxns[7])
-	require.Equal(t, updateSequencerFeeVaultProxySender, common.Address{})
-	require.Equal(t, updateSequencerFeeVaultProxySource.SourceHash(), updateSequencerFeeVaultProxy.SourceHash())
-	require.NotNil(t, updateSequencerFeeVaultProxy.To())
-	require.Equal(t, *updateSequencerFeeVaultProxy.To(), predeploys.SequencerFeeVaultAddr)
-	require.Equal(t, uint64(50_000), updateSequencerFeeVaultProxy.Gas())
-	require.Equal(t, common.FromHex("3659cfe600000000000000000000000013963c74d9c62d31ce3fcec0d46f6430a74ea79e"), updateSequencerFeeVaultProxy.Data())
+	require.Equal(t, expectedEncodedArg, encodedArg)
+
+	// create deployment bytecode
+	deploymentBytecode := createDeploymentBytecode(common.FromHex(bytecode), common.HexToAddress(constructorArg))
+
+	// Validate deployment bytecode
+	expectedDeploymentBytecodeHex := bytecode + expectedEncodedArg
+	actualDeploymentBytecodeHex := "0x" + hex.EncodeToString(deploymentBytecode)
+	require.Equal(t, expectedDeploymentBytecodeHex, actualDeploymentBytecodeHex, "deployment bytecode mismatch")
+
+	// Validate the split between base bytecode and constructor argument
+	actualBytecode := actualDeploymentBytecodeHex[:len(bytecode)]
+	actualArg := actualDeploymentBytecodeHex[len(bytecode):]
+	require.Equal(t, bytecode, actualBytecode, "base bytecode invalid")
+	require.Equal(t, expectedEncodedArg, actualArg, "constructor argument invalid")
+}
+
+// Function to encode constructor arguments
+func encodeConstructorArg(arg string) (string, error) {
```

Comment:
How about moving this function to `kromampt_upgrade_transaction.go` and share it with this test file?